src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ValidateVariants.java

```diff
@@ -286,6 +315,10 @@ public Object onTraversalSuccess() {
                         uncoveredIntervals.iterator().next());
                 throwOrWarn(e);
             }
+            if (FAIL_ON_OVERLAP && sawOverlap) {
+                final UserException e = new UserException("This GVCF contained overlapping reference blocks.  See log above for details.");
+                throwOrWarn(e);
```

Comment:
I think this should follow the same pattern as the above uncovered intervals check where the error at the end reports where one of these overlaps exists. You could make sawOverlap hold onto the first two offending variants etc...

Reply:
I think is best to fail early.